seacrowd/sea_datasets/mywsl2023/mywsl2023.py

```diff
@@ -0,0 +1,172 @@
+
+_SPLITS = ["train", "test"]
+
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+
+class MyWsl2023(datasets.GeneratorBasedBuilder):
+    """This dataset contains pictures of hand gestures corresponding to ten commonly-used Malaysian Sign Language (XML) words."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    subset_id = _DATASETNAME
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{subset_id}_source",
+            version=SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=subset_id,
+        )
+    ]
+
+    seacrowd_schema_config: list[SEACrowdConfig] = []
+
+    for seacrowd_schema in _SUPPORTED_SCHEMA_STRINGS:
+
+        seacrowd_schema_config.append(
+            SEACrowdConfig(
+                name=f"{subset_id}_{seacrowd_schema}",
+                version=SEACROWD_VERSION,
+                description=f"{_DATASETNAME} {seacrowd_schema} schema",
+                schema=f"{seacrowd_schema}",
+                subset_id=subset_id,
+            )
+        )
+
+    BUILDER_CONFIGS.extend(seacrowd_schema_config)
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "image_paths": datasets.Sequence(datasets.Value("string")),
+                    "texts": datasets.Value("string"),
+                }
+            )
+
+        elif self.config.schema == f"seacrowd_{str(TASK_TO_SCHEMA[Tasks.SIGN_LANGUAGE_RECOGNITION]).lower()}":
+            features = schemas.image_text_features(label_names=[])
+
+        else:
+            raise ValueError(f"Invalid config: {self.config.name}")
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        split_generators = []
+
+        path = dl_manager.download_and_extract(_URLS[_DATASETNAME])
+
+        for split in _SPLITS:
+            split_generators.append(
+                datasets.SplitGenerator(
+                    name=split,
+                    gen_kwargs={
+                        "path": os.path.join(path, "MyWSL2023 RAW DATA", split),
+                    },
+                )
+            )
+
+        return split_generators
+
+    def _generate_examples(self, path: str) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+
+        idx = 0
+        image_folder_paths = [os.path.join(path, folder) for folder in os.listdir(path)]
+
+        for image_folder_path in image_folder_paths:
```

Comment:
Let's use enumerate for updating `idx`.

Reply:
I might misunderstood this, but I manually increment the idx so that the value is globally unique for different image_folder_path. Using enumerate will not give global unique value. Please correct me if I'm wrong. @akhdanfadh

Reply:
It will. `idx` in enumerate will updated for each `image_folder_path`, thus unique. Already tested with below code.

```python
for idx, image_folder_path in enumerate(image_folder_paths):
    ... # remove all idx increment
```

Reply:
I think both options are correct if every row of datapoints contains all of the image within a folder. Had it been iterated per image, the approach would've been different